stem/model.py

```diff
@@ -191,6 +172,186 @@ def generate_straight_track(self, sleeper_distance: float, n_sleepers: int, rail
+        ]
+
+        soil_equivalent_line_ids = [ids[0] for ids in soil_equivalent_lines]
+
+        self.gmsh_io.add_physical_group(soil_equivalent_name, 1, soil_equivalent_line_ids)
+        soil_equivalent_parameters = extended_soil_parameters
+
+        soil_equivalent_part = BodyModelPart(soil_equivalent_name)
+        soil_equivalent_part.get_geometry_from_geo_data(self.gmsh_io.geo_data, soil_equivalent_name)
+        soil_equivalent_part.material = StructuralMaterial(name=soil_equivalent_name,
+                                                           material_parameters=soil_equivalent_parameters)
+        # can only move in the vertical direction
+        constrain_list = [True, True, True]
+        constrain_list[VERTICAL_AXIS] = False
+        soil_equivalent_part.parameters = DisplacementConstraint(active=constrain_list,
+                                                                 is_fixed=constrain_list,
+                                                                 value=[0, 0, 0])
+        self.body_model_parts.append(soil_equivalent_part)
+        # add bottom points fixed
+        constraint_model_soil_equivalent_name = f"constraint_{soil_equivalent_name}"
+        constraint_model_soil_equivalent_part = ModelPart(f"constraint_{soil_equivalent_name}")
+        constraint_model_soil_equivalent = DisplacementConstraint(active=[True, True, True],
+                                                                  is_fixed=[True, True, True],
+                                                                  value=[0, 0, 0])
+        constraint_model_soil_equivalent_part.parameters = constraint_model_soil_equivalent
+        constraint_model_soil_equivalent_part_settings = {
+            constraint_model_soil_equivalent_name: {
+                "coordinates": soil_equivalent_bottom,
+                "ndim": 0
+            }
+        }
+        self.gmsh_io.generate_geometry(constraint_model_soil_equivalent_part_settings, "")
+
+        constraint_model_soil_equivalent_part.get_geometry_from_geo_data(self.gmsh_io.geo_data,
+                                                                         constraint_model_soil_equivalent_name)
+
+        self.process_model_parts.append(constraint_model_soil_equivalent_part)
+
+    def get_points_outside_soil_volume(
+            self, model_part_name) -> Tuple[List[Union[int, Any]], List[Union[Sequence[float], Any]]]:
+        """
+        Get the points of the model part that are outside the volume of the model part.
+
+        Args:
+            - model_part_name (str): The name of the model part to check the points
+
+        Returns:
+            - List[int]: The ids of the points that are outside the volume of the model part.
+            - List[List[float]]: The coordinates of the points that are outside the volume of the model part.
+
+        """
+        # get bbox of the soil model parts
+        min_coords, max_coords = self.get_bounding_box_soil()
```

Comment:
you can use the gmsh.getBoundingBox function in gmsh i think. For this it is required to also add a function in gmsh_utils to call gmsh.GetBoundingBox

Reply:
I will make a separate issue about this so that when it is merged there we can update this

Reply:
I made a separate issue for this https://github.com/StemVibrations/STEM/issues/245